benchmark_runner/common/clouds/BareMetal/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+            self._install_ocp_version = self._get_installation_version()
+        openshift_version_data = self._install_ocp_version.split('.')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='ocp_release_image:',
+                                            value=f'quay.io\/openshift-release-dev\/ocp-release:{self._install_ocp_version}-x86_64')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='openshift_version:',
+                                            value=f'"{openshift_version_data[0]}.{openshift_version_data[1]}"')
+
+    def update_provision_config(self):
+        """
+        This method update provision config file
+        :return:
+        """
+        # update ssh config - must add it and also mount it
+        self._ssh.run(f"echo Host provision > /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'HostName {self._provision_ip} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'User {self._user} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'IdentityFile {self._container_private_key_path} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'StrictHostKeyChecking no >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveInterval 30 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveCountMax 5 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"chmod 600 /{self._user}/.ssh/config")
+
+    @logger_time_stamp
+    def run_ocp_installer(self):
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        self.update_provision_config()
+        logger.info(f'Starting OCP assisted installer, Start time: {datetime.now().strftime(datetime_format)}')
+        # Must add -t otherwise remote ssh of ansible will not end
+        self._ssh.run(f"ssh -t provision \"{self._install_ocp_cmd()}\" ")
+        logger.info(f'OpenShift cluster {self._get_installation_version()} version is installed successfully, End time: {datetime.now().strftime(datetime_format)}')
+
+    @logger_time_stamp
+    def verify_install_complete(self):
+        """
+        This method verify that installation complete
+        :return: True if installation success and raise exception if installation failed
+        """
+        complete = self._wait_for_install_complete()
+        if not complete:
+            raise Exception(f'Installation failed')
```

Comment:
Can you get any information from _wait_from_install_complete (such as messages from the installer)?  If so, please print at least the last 10 lines or so of that output.

Reply:
Fixed